test/cql-pytest/test_type_date.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+#############################################################################
+# Test involving the "date" column type.
+#############################################################################
+
+from util import unique_name, unique_key_int
+from cassandra.util import Date
+from cassandra.protocol import InvalidRequest
+
+import pytest
+
+@pytest.fixture(scope="module")
+def table1(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (p int PRIMARY KEY, d date)")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+# According to the Cassandra documentation, a date "can be input either as an
+# integer or using a date string."
+# The documentation further explains the (surprising) encoding of that
+# integer:
+#    "Values of the date type are encoded as 32-bit unsigned integers
+#     representing a number of days with “the epoch” at the center of the
+#     range (2^31). Epoch is January 1st, 1970"
+# So the number "123" doesn't mean 123 days since the epoch - it actually
+# means 2^31-123 days before the epoch.
+# The date string should be yyyy-mm-dd, and nothing else is allowed.
+# The following tests verify that these different initialization options
+# actually work.
+def test_type_date_from_int_unprepared(cql, table1):
+    p = unique_key_int()
+    cql.execute(f"INSERT INTO {table1} (p, d) VALUES ({p}, 123)")
+    ds = list(cql.execute(f"SELECT d from {table1} where p = {p}"))
+    assert len(ds)==1
+    # The Python driver returns a "date" type as a cassandra.util.Date
+    # type, and when that is converted into a number, it strangely results
+    # in the number of days since the epoch, where the epoch is 0.
+    # So 123 is converted into "-2^31+123" days (a negative number) since
+    # the epoch.
+    assert ds[0].d == -2**31 + 123
+
+def test_type_date_from_int_prepared(cql, table1):
+    p = unique_key_int()
+    stmt = cql.prepare(f"INSERT INTO {table1} (p, d) VALUES (?, ?)")
+    cql.execute(stmt, [p, 123])
+    ds = list(cql.execute(f"SELECT d from {table1} where p = {p}"))
+    assert len(ds)==1
```

Comment:
Don't we have some PEP8 check that'd eliminate such "glued" expressions? (In this case, by "glued" I mean that there's no whitespace between `==`).